app/selectors/accountsController.test.ts

```diff
@@ -186,3 +189,114 @@ describe('Accounts Controller Selectors', () => {
     });
   });
 });
+
+describe('Bitcoin Account Selectors', () => {
+  const MOCK_BTC_MAINNET_ADDRESS = 'bc1qkv7xptmd7ejmnnd399z9p643updvula5j4g4nd';
+  const MOCK_BTC_TESTNET_ADDRESS = 'tb1q63st8zfndjh00gf9hmhsdg7l8umuxudrj4lucp';
+
+  const MOCK_BTC_MAINNET_UUID = createMockUuidFromAddress(
+    MOCK_BTC_MAINNET_ADDRESS,
+  );
+
+  const MOCK_BTC_TESTNET_UUID = createMockUuidFromAddress(
+    MOCK_BTC_TESTNET_ADDRESS,
+  );
+
+  const btcMainnetAccount = createMockInternalAccount(
+    MOCK_BTC_MAINNET_ADDRESS,
+    'Bitcoin Account',
+    KeyringTypes.snap,
+    BtcAccountType.P2wpkh,
+  );
+
+  const btcTestnetAccount = createMockInternalAccount(
+    MOCK_BTC_TESTNET_ADDRESS,
+    'Bitcoin Testnet Account',
+    KeyringTypes.snap,
+    BtcAccountType.P2wpkh,
+  );
+
+  describe('hasCreatedBtcMainnetAccount', () => {
+    it('returns true when a BTC mainnet account exists', () => {
+      const state = {
+        engine: {
+          backgroundState: {
+            AccountsController: {
+              internalAccounts: {
+                accounts: {
+                  [MOCK_BTC_MAINNET_UUID]: btcMainnetAccount,
+                },
+                selectedAccount: MOCK_BTC_MAINNET_UUID,
+              },
+            },
+            KeyringController: MOCK_KEYRING_CONTROLLER,
+          },
+        },
+      } as RootState;
```

Comment:
Slight refactor proposal, what about having a function for this like:

```ts
function getStateWithAccount(account: InternalAccount) {
  return {
        engine: {
          backgroundState: {
            AccountsController: {
              internalAccounts: {
                accounts: {
                  [account.id]: account,
                },
                selectedAccount: account.id,
              },
            },
            KeyringController: MOCK_KEYRING_CONTROLLER,
          },
        },
      } as RootState;
}
```

This would be more DRY:

```suggestion
      const state = getStateWithAccount(btcMainnetAccount);
```

WDYT?

Reply:
Love it. the above tests are using a pre defined mock so it was not as needed for them but in this case it makes perfect sense. I have made this change in the latest commit.